README.md

```diff
@@ -18,40 +20,48 @@ The different parts of Cloudflow work in unison to dramatically accelerate your
 
 ## Basic Concepts
 
-// TODO Add an illustration here
+<p>
+<img src="./docs/images/apps.png" width="600">
+
+<i>Basic components of a Cloudflow Application</i>
+</p>
 
 As we discussed above, Cloudflow allows developers to quickly build and deploy distributed stream processing applications by breaking such applications into smaller stream processing units called `Streamlets`.
 Each `Streamlet` represents an independent stream processing component that implements a self-contained stage of the application logic.
 `Streamlets` let you break down your application into logical pieces that communicate with each other in a streaming fashion to accomplish an end to end goal.
 
-`Streamlets` can be composed into larger systems using _blueprints_, which specify how `Streamlets` can be connected together to form a topology.
+`Streamlets` can be composed into larger systems using _blueprints_, which specify how `Streamlets` can be connected together to form a topology. The topology formed defines a DAG.
```

Comment:
Why do you want to add this? (It's also not exactly true, BTW, we do not stop anyone from creating a cycle)

Reply:
I would prefer to leave this addition out.

Reply:
That is the correct definition of the topology, if you have one I would like to add it here.